workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -94,16 +115,29 @@ def _restore_files_to_manager(self):
 
     def _restore_db(self, es, postgres, snapshot_version):
         ctx.logger.info('Restoring database')
-        if snapshot_version >= self._V_4_0_0:
+        if snapshot_version >= V_4_0_0:
             postgres.restore(self._tempdir)
         else:
-            postgres.create_clean_db()
+            if self._should_clean_old_db_for_3_x_snapshot():
```

Comment:
Added this clause - now cleaning the DB when restoring 3.x snapshots only if working in community edition, the DB is dirty and the `force` flag was passed